compare_template_text/normalize_template_text.py

```diff
@@ -0,0 +1,65 @@
+import re
+import sys
+
+"""This class normalizes the template by implementing the Omitable and Replaceable text 
+normalizations. """
+
+class NormalizeTemplate:
+    
+    def __init__(self,text_string,template_string):
+        self.text_string = text_string
+        self.template_string = template_string
+                
+    def normalize_template(self):
+        self.remove_bulletted_text()
+        self.remove_omitable_text()
+        self.remove_repeating_chars()
+    
+    def return_normalized_template(self):
+        
+        """ Returns the Normalized template after normalization techniques. """
+        print(self.template_string)
+        return self.template_string
+    
+    def return_normalized_text(self):
+        """ Returns the normalized text after normalization techniques. """
+        
+        print(self.text_string)
+        return self.text_string
+    
+    def remove_omitable_text(self):
+        """ Removes text from the template and the text that is optional. So, the text 
+        which is optional and present matches and gets replaced by '' while the text that doesn't match will 
+        remain as it is thus giving an unmatch. """
+         
+        lists = re.findall('(?<=\<\<beginoptional\>\>).*?(?=\<\<endoptional\>\>)',
+                           self.template_string)
+        # print(lists)
+        for x in lists:
+            if(x.startswith('*')):
+                x = x[1:]
+
+            if(x.endswith('*')):
+                x = x[:-1]
+
+            self.text_string = self.text_string.replace(x,'')
```

Comment:
Will this remove more than we want?  For example, lets say have have a template:

```
This is an optional <beginOptional>an<endOptional>.
```
I believe the above algorithm would result in:
```
This is optional.
```
where it should result in 
```
This is an optional.
``` 
(note the extra `an` removed)

Reply:
@goneall Yes even I thought that it will remove more than required but I looked on some templates and the optional tags have mainly the heading of the license which isn't repeated. But I will still try to look upon the case. Actually the optional tags may appear both at the beginning and at the end so it's not possible to match it with the word just before or after the tags. The position of the tags is also not the same in the text and the template. If it's all right can I look upon this issue after completing the basic functionalities of the project?

Reply:
@anshuldutt21 I think this is an important to resolve since it may lead to false positives.  Why don't you open an issue against it so we don't forget to fix it.  I can be addressed later if that is more efficient for you.  I can think of solutions to the matching issues - feel free to ping me on gitter if you want to discuss real-time.

Reply:
@goneall All right I will open an issue.